lib/SIL/SILCoverageMap.cpp

```diff
@@ -40,6 +41,13 @@ SILCoverageMap::create(SILModule &M, StringRef Filename, StringRef Name,
   CM->MappedRegions = M.allocateCopy(MappedRegions);
   CM->Expressions = M.allocateCopy(Expressions);
 
+  // Assert that this coverage map is unique.
+  assert(llvm::none_of(M.coverageMaps,
+                       [&](const SILCoverageMap &OtherCM) {
+                         return OtherCM.PGOFuncName == CM->PGOFuncName;
+                       }) &&
+         "Duplicate coverage mapping for function");
+
   M.coverageMaps.push_back(CM);
```

Comment:
Why not use a DenseMap instead? The assertion is O(n)...

Reply:
I thought the PR would be cleaner without this change. I'm planning on doing this as a follow-up, but can do it now if you'd prefer.

Reply:
I'm going to look into the fallout from the source compatibility suite now. Let me know if you think I should make the DenseMap change in this PR (I suppose it'd have to be MapVector for deterministic printing).